File: plugin/src/org/jetbrains/haskell/HaskellViewProvider.java
package org.jetbrains.haskell;

import com.intellij.lang.Language;
import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.psi.PsiManager;
import com.intellij.psi.SingleRootFileViewProvider;
import org.jetbrains.annotations.NotNull;

public class HaskellViewProvider extends SingleRootFileViewProvider {

  public HaskellViewProvider(@NotNull PsiManager manager, @NotNull VirtualFile file) {
   super(manager, file);
  }

  public HaskellViewProvider(@NotNull PsiManager manager, @NotNull VirtualFile virtualFile, final boolean physical) {
    super(manager, virtualFile, physical);
  }

  public HaskellViewProvider(@NotNull PsiManager manager, @NotNull VirtualFile virtualFile, final boolean physical, @NotNull Language language) {
    super(manager, virtualFile, physical, language);
  }

  public boolean supportsIncrementalReparse(final Language rootLanguage) {
    return false;
  }

}

File: plugin/src/org/jetbrains/haskell/HaskellViewProvider.kt
package org.jetbrains.haskell

import com.intellij.lang.Language
import com.intellij.openapi.vfs.VirtualFile
import com.intellij.psi.PsiManager
import com.intellij.psi.SingleRootFileViewProvider

public class HaskellViewProvider(manager: PsiManager,
                                 virtualFile: VirtualFile,
                                 eventSystemEnabled: Boolean,
                                 language: Language) :
        SingleRootFileViewProvider(manager, virtualFile, eventSystemEnabled, language) {

    override fun supportsIncrementalReparse(rootLanguage: Language): Boolean {
        return false
    }

}
